codex/sales/slotqueue.nim

```diff
@@ -245,6 +263,13 @@ proc push*(self: SlotQueue, item: SlotQueueItem): ?!void =
     self.queue.del(self.queue.size - 1)
 
   doAssert self.queue.len <= self.queue.size - 1
+
+  # when slots are pushed to the queue, the queue should be unpaused if it was
+  # paused
+  if self.paused:
+    trace "unpausing queue after new slots pushed"
+  self.unpause()
```

Comment:
While I guess it can be fired everytime, shouldn't it be intended under the `if` above?
```suggestion
    self.unpause()
```

Reply:
Yep, it doesn't matter but can update for optics.